### PR TITLE
Unlock Account

### PR DESCRIPTION
**Pull request by jaredperreault-okta**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,9 @@
   - Removes signOut option `clearTokensAfterRedirect`
   - Adds signOut option `clearTokensBeforeRedirect` (default: `false`) to remove local tokens before logout redirect happen
 - [#1057](https://github.com/okta/okta-auth-js/pull/1057) Strict checks are now enabled in the Typescript compiler options. Some type signatures have been changed to match current behavior.
+- [#1066](https://github.com/okta/okta-auth-js/pull/1066) Extends type `CookieOptions` from `js-cookie` v3
```
Comment (shuowu-okta) [inline CHANGELOG.md:15]: reverted by Aaron

Comment (jaredperreault-okta) [reply CHANGELOG.md:15]: 🤔 rebase error?

Comment (shuowu-okta) [reply CHANGELOG.md:15]: maybe because you squashed all commits into one before rebasing.

Comment (jaredperreault-okta) [reply CHANGELOG.md]: fixed

File: lib/idx/remediate.ts [hunk 2]
```diff
@@ -211,11 +197,55 @@ export async function remediate(
         }
         return remediate(idxResponse, valuesWithoutExecutedAction, options); // recursive call
       }
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      else if (typeof action ==='string' && action in remediators!) {
```
Comment (shuowu) [inline lib/idx/remediate.ts:201]: nit: `actions` is in type string[], no type check is needed here.

File: lib/idx/remediate.ts [hunk 3]
```diff
@@ -211,11 +197,55 @@ export async function remediate(
...
+        const remediation = neededToProceed.find(({ name }) => name === action);
+        if (!remediation) {
+          continue;
+        }
+        foundRemediationCandidate = true;   // signal remediation(s) were passed as actions
```
Comment (shuowu) [inline lib/idx/remediate.ts:206]: here action means remediation that does not need any data from user interaction, how about return idx.proceed(name) directly?

File: lib/idx/remediate.ts [hunk 4]
```diff
@@ -211,11 +197,55 @@ export async function remediate(
...
+
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        const T = remediators![action];
+        const remediatorCandidate = new T(remediation, values);
+        if (remediatorCandidate.canRemediate(flow)) {
```
Comment (shuowu) [inline lib/idx/remediate.ts:211]: action like remediation should have no input data, which mean it should always be able to remediate.

File: lib/idx/remediate.ts [hunk 5]
```diff
@@ -211,11 +197,55 @@ export async function remediate(
...
+          remediator = remediatorCandidate;
+        }
+      }
+    }
+
+    // if remediations were passed as actions, but were unable to proceed/remediate, throw
+    if (foundRemediationCandidate && !remediator) {
```
Comment (shuowu) [inline lib/idx/remediate.ts:218]: this thrown error won't be needed if we can return when the action is found.

File: lib/idx/remediate.ts (hunk 2 above)
Comment (jaredperreault-okta) [reply lib/idx/remediate.ts:201]: Ha. Still getting used to TS

File: lib/idx/remediators/Base/Remediator.ts [hunk 6]
```diff
@@ -81,6 +81,7 @@ export class Remediator {
   }
 
   // Override this method to provide custom check
+  /* eslint-disable-next-line no-unused-vars, @typescript-eslint/no-unused-vars */
```
Comment (aarongranick-okta) [inline lib/idx/remediators/Base/Remediator.ts:84]: probably don't need this now

File: samples/templates/express-embedded-auth-with-sdk/web-server/routes/unlock-account.js [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+  });
+
+  renderTemplate(req, res, 'unlock-account', {
+    action: '/unlock-account'
+  });
+});
+
+router.post('/unlock-account', async (req, res, next) => {
+  const authClient = getAuthClient(req);
+  const transaction = await authClient.idx.unlockAccount({});
```
Comment (aarongranick-okta) [inline samples/templates/express-embedded-auth-with-sdk/web-server/routes/unlock-account.js:37]: should do a try/catch around the idx operations, unless sure that the error will be caught higher up. there is a possibility these methods will throw

**Review by aarongranick-okta — APPROVED** [review]

**Pull request closed without merging**